src/main/java/net/minestom/server/inventory/AbstractInventory.java

```diff
@@ -0,0 +1,362 @@
+     *
+     * @param itemStacks items to be checked
+     * @return true if all the items can be fully taken from the inventory, false otherwise
+     */
+    public boolean canTakeItemStacks(@NotNull List<ItemStack> itemStacks) {
+        return itemStacks.stream().allMatch(this::canTakeItemStack);
+    }
+
+    public synchronized void replaceItemStack(int slot, @NotNull UnaryOperator<@NotNull ItemStack> operator) {
+        var currentItem = getItemStack(slot);
+        setItemStack(slot, operator.apply(currentItem));
+    }
+
+    /**
+     * Clears the inventory and send relevant update to the viewer(s).
+     */
+    public synchronized void clear() {
+        // Clear the item array
+        Arrays.fill(itemStacks, ItemStack.AIR);
+        // Send the cleared inventory to viewers
+        update();
+    }
+
+    public abstract void update();
+
+    /**
+     * Gets the {@link ItemStack} at the specified slot.
+     *
+     * @param slot the slot to check
+     * @return the item in the slot {@code slot}
+     */
+    public @NotNull ItemStack getItemStack(int slot) {
+        return itemStacks[slot];
+    }
+
+    /**
+     * Gets all the {@link ItemStack} in the inventory.
+     * <p>
+     * Be aware that the returned array does not need to be the original one,
+     * meaning that modifying it directly may not work.
+     *
+     * @return an array containing all the inventory's items
+     */
+    public @NotNull ItemStack[] getItemStacks() {
+        return itemStacks.clone();
+    }
+
+    /**
+     * Gets the size of the inventory.
+     *
+     * @return the inventory's size
+     */
+    public int getSize() {
+        return size;
+    }
+
+    /**
+     * Gets the size of the "inner inventory" (which includes only "usable" slots).
+     *
+     * @return inner inventory's size
+     */
+    public int getInnerSize() {
+        return getSize();
+    }
+
+    /**
+     * Gets all the {@link InventoryCondition} of this inventory.
+     *
+     * @return a modifiable {@link List} containing all the inventory conditions
+     */
+    public @NotNull List<InventoryCondition> getInventoryConditions() {
+        return inventoryConditions;
+    }
+
+    /**
+     * Adds a new {@link InventoryCondition} to this inventory.
+     *
+     * @param inventoryCondition the inventory condition to add
+     */
+    public void addInventoryCondition(@NotNull InventoryCondition inventoryCondition) {
+        this.inventoryConditions.add(inventoryCondition);
+    }
+
+    /**
+     * Places all the items of {@code itemStacks} into the internal array.
+     *
+     * @param itemStacks the array to copy the content from
+     * @throws IllegalArgumentException if the size of the array is not equal to {@link #getSize()}
+     * @throws NullPointerException     if {@code itemStacks} contains one null element or more
+     */
+    public void copyContents(@NotNull ItemStack[] itemStacks) {
```

Comment:
Is there a use case for this? I feel like `getContents` is probably good enough unless there's a specific use case for this method I'm missing.

Reply:
In my case, retrieving all player items from a database